sympy/integrals/tests/test_integrals.py

```diff
@@ -20,6 +21,42 @@
 n = Symbol('n', integer=True)
 f = Function('f')
 
+def test_principal_value():
+
+    g = 1 / x
+    assert Integral(g, (x, -oo, oo)).principal_value() == nan
+    assert Integral(g, (y, -oo, oo)).principal_value() == oo*sign(1/x)
+    raises(ValueError, lambda: Integral(g, (x)).principal_value())
+    raises(ValueError, lambda: Integral(g).principal_value())
+
+    l = 1 / ((x**3) - 1)
+    assert Integral(l, (x, -oo, oo)).principal_value() == nan
```

Comment:
@jksuom
it is giving nan because of the reason which I have explained in the documentation's example ( line no. 1268-1274 )

this is the explaination :-
```
This happens so because, the method principal_value returns -oo for Integral from -oo to 0
and +oo for Integral from 0 to oo. While adding both the Integrals i.e. -oo + oo we do not
get 0 (as we expect) but nan since sympy realises that -oo and +oo aren't actual numbers that
can be added/subtracted.
```
does this make sense?